src/RealtimeClient.ts

```diff
@@ -317,6 +321,76 @@ export default class RealtimeClient {
     })
   }
 
+  /**
+   * Creates a private channel to be used with the provided name.
+   *
+   * @param name Channel name to create
+   */
+  createPrivateChannel(name: string): Promise<string> {
+    const url = `${this.httpEndpoint}/channels?apikey=${this.apiKey}`
+    return this.fetch(url, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ name }),
+    }).then((response) => {
+      if (!response.ok && response.status !== 200) {
+        throw new Error(response.statusText)
+      }
+      return name
+    })
+  }
+
+  /**
+   * Deletes a private channel
+   *
+   * @param name Channel name to delete.
+   */
+  deletePrivateChannel(name: string): Promise<boolean> {
+    const url = `${this.httpEndpoint}/channels/${name}?apikey=${this.apiKey}`
+    return this.fetch(url, {
+      method: 'DELETE',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ name }),
+    }).then((response) => {
+      if (!response.ok && response.status !== 202) {
+        throw new Error(response.statusText)
+      }
+      return true
+    })
+  }
+
+  /**
+   * Update a private channel
+   *
+   * @param name Channel name to update.
+   * @param new_name New channel name.
+   */
+  updatePrivateChannel(name: string, new_name: string): Promise<string> {
+    const url = `${this.httpEndpoint}/channels/${name}?apikey=${this.apiKey}`
+    return this.fetch(url, {
+      method: 'PUT',
```

Comment:
@filipecabaco isn't this technically a `PATCH` since you're only updating the name? Doesn't `Channel.changeset` also contain `check` attr?

Reply:
we accept bot PUT and PATCH in the endpoints so we can go with PATCH to be more correct

Reply:
check is a non important field as it's used for checking only if the user has write policies and not to be used by users. we will explain that in the docs probably in the architecture side

Reply:
non important aka internal